vic/drivers/classic/src/parse_output_info.c

```diff
@@ -59,96 +58,106 @@ parse_output_info(FILE                  *gp,
-                }
-                else {
-                    if (strcasecmp("OUT_TYPE_USINT", typestr) == 0) {
-                        type = OUT_TYPE_USINT;
-                    }
-                    else if (strcasecmp("OUT_TYPE_SINT", typestr) == 0) {
-                        type = OUT_TYPE_SINT;
+        *out_data_files = calloc(options.Noutfiles,
+                                 sizeof(*(*out_data_files)));
+        init_output_list(out_data, false, "%.4f", OUT_TYPE_FLOAT, 1);
+
+        // PRT_SNOW_BAND is ignored if N_OUTFILES has been specified
+        options.PRT_SNOW_BAND = false;
+
+        while (!feof(gp)) {
+            if (cmdstr[0] != '#' && cmdstr[0] != '\n' && cmdstr[0] != '\0') {
+                sscanf(cmdstr, "%s", optstr);
+
+                if (strcasecmp("OUTFILE", optstr) == 0) {
+                    outfilenum++;
+                    if (!options.Noutfiles) {
+                        log_err("in global param file: \"N_OUTFILES\" must be "
+                                "specified before you can specify \"OUTFILE\".");
                     }
-                    else if (strcasecmp("OUT_TYPE_FLOAT", typestr) == 0) {
-                        type = OUT_TYPE_FLOAT;
+                    if (outfilenum >= (short int)options.Noutfiles) {
+                        log_err("number of output files specified in N_OUTFILES "
+                                "(%zu) is less than actual number of output files "
+                                "defined in the global param file.",
+                                options.Noutfiles);
                     }
-                    else if (strcasecmp("OUT_TYPE_DOUBLE", typestr) == 0) {
-                        type = OUT_TYPE_DOUBLE;
+                    sscanf(cmdstr, "%*s %s", (*out_data_files)[outfilenum].prefix);
+
+                    // determine how many variable will be in this file before
+                    // allocating (GH: 209)
+                    (*out_data_files)[outfilenum].nvars = count_outfile_nvars(gp);
+
+                    (*out_data_files)[outfilenum].varid =
+                        calloc((*out_data_files)[outfilenum].nvars,
+                               sizeof(*((*out_data_files)[outfilenum].varid)));
+                    outvarnum = 0;
+                }
+                else if (strcasecmp("OUTVAR", optstr) == 0) {
+                    if (outfilenum < 0) {
+                        log_err("Error in global param file: \"OUTFILE\" must be "
+                                "specified before you can specify \"OUTVAR\".");
                     }
-                    else {
+                    strcpy(format, "");
+                    strcpy(typestr, "");
+                    strcpy(multstr, "");
+                    sscanf(cmdstr, "%*s %s %s %s %s", varname, format, typestr,
+                           multstr);
+                    if (strcasecmp("", format) == 0) {
+                        strcpy(format, "*");
```

Comment:
That makes sense.  I'm not going to change this though since I didn't actually change this line.  This is a github diff thing.